server/protocols/webadmin/webadmin-cassandra-data/src/main/java/org/apache/james/webadmin/service/CassandraMappingsService.java

```diff
@@ -0,0 +1,50 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.service;
+
+import javax.inject.Inject;
+
+import org.apache.james.rrt.cassandra.CassandraMappingsSourcesDAO;
+import org.apache.james.rrt.cassandra.migration.MappingsSourcesMigration;
+import org.apache.james.task.Task;
+import org.apache.james.webadmin.dto.CassandraActionMappings;
+
+public class CassandraMappingsService {
+    private final MappingsSourcesMigration mappingsSourcesMigration;
+    private final CassandraMappingsSourcesDAO cassandraMappingsSourcesDAO;
+
+    @Inject
+    public CassandraMappingsService(MappingsSourcesMigration mappingsSourcesMigration,
+                                    CassandraMappingsSourcesDAO cassandraMappingsSourcesDAO) {
+        this.mappingsSourcesMigration = mappingsSourcesMigration;
+        this.cassandraMappingsSourcesDAO = cassandraMappingsSourcesDAO;
+    }
+
+    public Task performAction(CassandraActionMappings.Action action) {
+        if (CassandraActionMappings.Action.solveInconsistencies.equals(action)) {
+            return solveMappingsSourcesInconsistencies();
+        }
+        throw new IllegalArgumentException(action + " is not a valid action");
```

Comment:
If it's not valid, why did you build it in the first place ?
You probably mean that you can't handle the Action.

Reply:
Totally right, the parse mathod takes care of it already actually